lib/usd/ui/layerEditor/saveLayersDialog.cpp

```diff
@@ -0,0 +1,414 @@
+    std::string   newRoot;
+    SessionState* rootSessionState = nullptr;
+
+    _newPaths.clear();
+    _problemLayers.clear();
+    _emptyLayers.clear();
+
+    for (i = 0, count = _rowsLayout->count(); i < count; ++i) {
+        auto row = dynamic_cast<SaveLayerPathRow*>(_rowsLayout->itemAt(i)->widget());
+        auto item = row ? row->_treeItem : nullptr;
+        if (nullptr == item)
+            continue;
+
+        QString path = row->pathToSaveAs();
+        if (!path.isEmpty()) {
+            pxr::SdfFileFormat::FileFormatArguments formatArgs;
+            formatArgs["format"] = row->usdFormatTag().toStdString();
+
+            auto sessionState = item->parentModel()->sessionState();
+            auto sdfLayer = item->layer();
+            auto qFileName = row->absolutePath();
+            auto sFileName = qFileName.toStdString();
+
+            sdfLayer->Export(sFileName, "", formatArgs);
+
+            if (item->isRootLayer()) {
+                newRoot = sFileName;
+                rootSessionState = sessionState;
+            } else {
+                // now replace the layer in the parent
+                auto parentItem = item->parentLayerItem();
+                auto newLayer = SdfLayer::FindOrOpen(sFileName);
+                if (newLayer) {
+                    bool setTarget = item->isTargetLayer();
+                    parentItem->layer()->GetSubLayerPaths().Replace(
+                        sdfLayer->GetIdentifier(), newLayer->GetIdentifier());
+                    if (setTarget) {
+                        sessionState->stage()->SetEditTarget(newLayer);
+                    }
+
+                    _newPaths.append(QString::fromStdString(sdfLayer->GetDisplayName()));
+                    _newPaths.append(qFileName);
+                } else {
+                    _problemLayers.append(QString::fromStdString(sdfLayer->GetDisplayName()));
+                    _problemLayers.append(qFileName);
+                }
+            }
+        } else {
+            _emptyLayers.append(row->layerDisplayName());
+        }
+    }
+
+    if (rootSessionState) {
+        rootSessionState->rootLayerPathChanged(newRoot);
+    }
+
+    accept();
+}
+
+void SaveLayersDialog::onCancel() { reject(); }
+
+bool SaveLayersDialog::okToSave()
+{
+    int         i, count;
+    QStringList existingFiles;
+
+    for (i = 0, count = _rowsLayout->count(); i < count; ++i) {
+        auto row = dynamic_cast<SaveLayerPathRow*>(_rowsLayout->itemAt(i)->widget());
+        auto item = row ? row->_treeItem : nullptr;
+        if (nullptr == item)
+            continue;
+
+        QString path = row->pathToSaveAs();
+        if (!path.isEmpty()) {
+            QFileInfo fInfo(path);
+            if (fInfo.exists()) {
+                existingFiles.append(path);
+            }
+        }
+    }
+
+    if (!existingFiles.isEmpty()) {
+        QString message
+            = QString("%1 file(s) already exist and will be overwritten.  Do you want to continue?")
```

Comment:
string not translated like the other. Also, that "i" and the "count" in the for loop should be declared in the "for"